Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs

```diff
@@ -265,25 +265,38 @@ public Task InvokeDidChangeConfigurationAsync(LSP.DidChangeConfigurationParams r
-            => _rpc == null ? Task.FromResult<object>(null) : _rpc.InvokeWithParameterObjectAsync<object>("textDocument/completion", request, cancellationToken);
-
-        public Task<TResult> InvokeWithParameterObjectAsync<TResult>(string targetName, object argument = null, CancellationToken cancellationToken = default)
-            => _rpc == null ? Task.FromResult(default(TResult)) : _rpc.InvokeWithParameterObjectAsync<TResult>(targetName, argument, cancellationToken);
+            => InvokeWithParametersAsync<object>("textDocument/completion", request, cancellationToken);
 
         public Task<object> InvokeTextDocumentSymbolsAsync(LSP.DocumentSymbolParams request, CancellationToken cancellationToken)
-                    => _rpc == null ? Task.FromResult(default(object)) : _rpc.InvokeWithParameterObjectAsync<object>("textDocument/documentSymbol", request, cancellationToken);
+            => InvokeWithParametersAsync<object>("textDocument/documentSymbol", request, cancellationToken);
 
         public Task<object> InvokeTextDocumentDefinitionAsync(LSP.TextDocumentPositionParams request, CancellationToken cancellationToken)
-                    => _rpc == null ? Task.FromResult(default(object)) : _rpc.InvokeWithParameterObjectAsync<object>("textDocument/definition", request, cancellationToken);
+            => InvokeWithParametersAsync<object>("textDocument/definition", request, cancellationToken);
 
         public Task<LSP.Location[]> InvokeReferencesAsync(LSP.ReferenceParams request, CancellationToken cancellationToken)
-            => _rpc == null ? Task.FromResult<LSP.Location[]>(null) : _rpc.InvokeWithParameterObjectAsync<LSP.Location[]>("textDocument/references", request, cancellationToken);
+            => InvokeWithParametersAsync<LSP.Location[]>("textDocument/references", request, cancellationToken);
 
         public Task<LSP.CompletionItem> InvokeResolveAsync(LSP.CompletionItem request, CancellationToken cancellationToken)
-            => _rpc == null ? Task.FromResult<LSP.CompletionItem>(null) : _rpc.InvokeWithParameterObjectAsync<LSP.CompletionItem>("completionItem/resolve", request, cancellationToken);
+            => InvokeWithParametersAsync<LSP.CompletionItem>("completionItem/resolve", request, cancellationToken);
         
         public Task<object> InvokeCommandAsync(LSP.ExecuteCommandParams request, CancellationToken cancellationToken)
-            => _rpc == null ? Task.FromResult<object>(null) : _rpc.InvokeWithParameterObjectAsync<object>("workspace/executeCommand", request, cancellationToken);
+            => InvokeWithParametersAsync<object>("workspace/executeCommand", request, cancellationToken);
+
+
+        private async Task<R> InvokeWithParametersAsync<R>(string request, object parameters, CancellationToken t) where R: class {
+            await _readyTcs.Task.ConfigureAwait(false);
+            if (_rpc != null) {
+                return await _rpc.InvokeWithParameterObjectAsync<R>(request, parameters, t);
```

Comment:
should we add another ConfigureAwait(false); to the `InvokeWithParameterObjectAsync` call?

Reply:
Sure that sounds good. Trying it now